packages/payment-detection/src/erc777/superfluid-detector.ts

```diff
@@ -21,15 +22,72 @@ export class SuperFluidPaymentDetector extends ReferenceBasedDetector<
     super(PaymentTypes.PAYMENT_NETWORK_ID.ERC777_STREAM, advancedLogic.extensions.erc777Stream);
   }
 
+  protected isSubsequentRequest(request: RequestLogicTypes.IRequest): boolean {
+    return !!request.extensions[this.paymentNetworkId].values.originalRequestId;
+  }
+
+  /**
+   * This returns the specific values we store for the ERC777 extensions.
+   * @param request The request we are processing
+   */
+  protected getSubsequentValues(
+    request: RequestLogicTypes.IRequest,
+  ): ExtensionTypes.PnStreamReferenceBased.ISubsequentRequestCreationParameters {
+    return request.extensions[this.paymentNetworkId].values;
+  }
+
+  /**
+   * With streaming requests we have a custom balance computation.
+   * It is possible to have only one streaming event, start stream,
+   * but have 10 requests being created each month for the same stream.
+   * For the original request, the first one, the balance computation is easy because we don't care about previous expected balance.
+   * For subsequent requests we must take in to consideration how many requests were before this one and then attribute the rest of the balance to this one.
+   * At this point we are not detecting overpayment because in case of overpayment we are considering that another request should have been created in the series.
+   * @param request The request we are calculating balance for
+   * @returns Balance object with balance value and payment/refund events
+   */
+  public async getBalance(
+    request: RequestLogicTypes.IRequest,
+  ): Promise<
+    PaymentTypes.IBalanceWithEvents<
+      PaymentTypes.IERC777PaymentEventParameters | PaymentTypes.IDeclarativePaymentEventParameters
+    >
+  > {
+    const totalBalance = await super.getBalance(request);
+    if (totalBalance.error) {
+      return totalBalance;
+    }
+    if (totalBalance.balance) {
+      const zeroBN = BigNumber.from(0);
+      let expectedPreviousBalance = zeroBN;
+      if (this.isSubsequentRequest(request)) {
+        const subrequestValues = this.getSubsequentValues(request);
+        expectedPreviousBalance = BigNumber.from(subrequestValues.recurrenceNumber).mul(
+          request.expectedAmount,
+        );
```

Comment:
Computing the balance on the recurrence number seems quite fragile, but I don't have a strong knowledge of superfluid. 

For instance, what ensures that this number is "valid" (ie, continuous)?

Reply:
Nothing strict for now. But we can ensure that each subsequent request has their recurrenceNumber as +1 from the previous one, we have previousRequestId. 
But that would add the complexity of fetching the previous request on each request creation and would add the dependency on requestLogic, currently advanced logic doesn't use request logic.

Another way is to take into consideration the expectedStartDate from the original requestId and calculate the number of months passed since then, to calculate the recurrence. In this case we depend on the original start date, is that more or less fragile?, and we also limit ourselves to 1 month duration.

I think this is enough for the MVP though

Reply:
@KolevDarko the recurring period is stored in the request so we are not limited to 1 month duration, it's just the UI which we have limited for MVP to 1 month duration. For a quick demo we should be able to set shorter period to see the recurrence happening during the demo.

Reply:
It should be remembered for after the MVP to not accept any recurrenceNumber but to only set it at creation from the recurrenceNumber of previousRequestId.